### PR TITLE
Feat/use neo-go 0.106.3 features

### LINKED ISSUE #2902: Upgrade to NeoGo 0.106.3
## Is your feature request related to a problem? Please describe.
I'm always frustrated when we're using outdated NeoGo. https://github.com/nspcc-dev/neo-go/releases/tag/v0.106.3 is here.

## Describe the solution you'd like
Upgrade. Use new APIs where appropriate.

### PR DESCRIPTION
**Pull request by carpawell**
neofs-adm is not touched to save mental health (https://github.com/nspcc-dev/neofs-node/issues/2905 was left). Closes #2902.

### CODE REVIEW

File: pkg/morph/client/notary.go [hunk 1]
```diff
@@ -209,7 +210,12 @@ func (c *Client) depositNotary(amount fixedn.Fixed8, till int64) (res util.Uint2
 			zap.Int64("expire_at", till),
 			zap.Uint32("vub", vub),
 			zap.Error(err))
-		return util.Uint256{}, nil
+		return nil
+	}
+
+	_, err = c.rpcActor.WaitSuccess(txHash, vub, err)
+	if err != nil {
+		return fmt.Errorf("waiting for %s TX to be persisted: %w", txHash.Reverse(), err)
```
Comment (cthulhu-rider) [inline pkg/morph/client/notary.go:218]:
```suggestion
		return fmt.Errorf("waiting for %s TX to be persisted: %w", txHash.StringLE(), err)
```
to clarify endianness

Comment (cthulhu-rider) [inline pkg/morph/client/notary.go:218]: can `vub` be useful in error?

**Review by cthulhu-rider — COMMENTED** [review]
lux gut

File: pkg/morph/client/notary.go (hunk 1 above)
Comment (carpawell) [reply pkg/morph/client/notary.go:218]: why not

Comment (carpawell) [reply pkg/morph/client/notary.go:218]: made additional commit

File: pkg/innerring/notary.go [hunk 2]
```diff
@@ -39,67 +37,36 @@ func (s *Server) depositMainNotary() (tx util.Uint256, err error) {
...
-	return awaitNotaryDepositInClient(ctx, s.mainnetClient, tx)
-}
-
-func (s *Server) awaitSideNotaryDeposit(ctx context.Context, tx util.Uint256) error {
-	return awaitNotaryDepositInClient(ctx, s.morphClient, tx)
-}
-
-func (s *Server) initNotary(ctx context.Context, deposit depositor, await awaiter, msg string) error {
-	tx, err := deposit()
+func (s *Server) initNotary(deposit depositor, msg string) error {
```
[the remark this answers is not in the document]
Comment (carpawell) [reply pkg/innerring/notary.go:63]: dont mind

**Review by cthulhu-rider — APPROVED** [review]

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**